### PR TITLE
Gating form micro adjustments

### PR DESCRIPTION
**Pull request by mzparacha**
## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/5458

## Description of Changes
- Updated API and frontend to support group topic updates
- Fixed chain ids and chain options in the requirements section
- Updated UI to show cosmos/eth base chain options based on the selected chain base
- Fix form style issues

## "How We Fixed It"
N/A

## Test Plan
- Go to the update group page with an admin account and verify that updating group topics work.
- Play around with the form requirements sections and verify it works correctly (i.e no style or breaking issues).

## Deployment Plan
N/A

## Other Considerations
N/A

### CODE REVIEW

File: packages/commonwealth/server/controllers/server_groups_methods/update_group.ts [hunk 1]
```diff
@@ -89,8 +108,52 @@ export async function __updateGroup(
...
+        },
+        {
+          where: {
+            id: {
+              [Op.in]: topicsToAssociate.map(({ id }) => id),
+            },
+          },
+          transaction,
+        }
+      );
```
Comment (rbennettcw) [inline packages/commonwealth/server/controllers/server_groups_methods/update_group.ts:133]: This allows duplicate group IDs to be added to `group_ids`, so the array might look like `[1, 2, 3, 4, 4, 4]`.

The code should ensure that no duplicate IDs are added.

**Review by rbennettcw — CHANGES_REQUESTED** [review]

File: packages/commonwealth/server/controllers/server_groups_methods/update_group.ts (hunk 1 above)
Comment (mzparacha) [reply packages/commonwealth/server/controllers/server_groups_methods/update_group.ts:133]: There is a code section below that which removed duplicate ids, also I am not able to reproduce


https://github.com/hicommonwealth/commonwealth/assets/51641047/22892c92-ba71-4907-b8fe-40adc30f57eb

File: packages/commonwealth/server/controllers/server_groups_methods/update_group.ts [hunk 2]
```diff
@@ -60,6 +63,22 @@ export async function __updateGroup(
     }
   }
 
+  let topicsToAssociate;
```
[the remark this answers is not in the document]
Comment (mzparacha) [reply packages/commonwealth/server/controllers/server_groups_methods/update_group.ts]: Added [here](https://github.com/hicommonwealth/commonwealth/pull/5499/commits/121b0268bc534296e4e22d81b51c4b12ca880840), thanks for the spot.

File: packages/commonwealth/server/controllers/server_groups_methods/update_group.ts (hunk 1 above)
Comment (rbennettcw) [reply packages/commonwealth/server/controllers/server_groups_methods/update_group.ts:134]: In the `topics` array, you're probably passing in a group that doesn't exist.

Try a group that exists. E.g. if the group ID is 1234, then set `topics` to `1234`, and invoke the PUT request multiple times.

Comment (mzparacha) [reply packages/commonwealth/server/controllers/server_groups_methods/update_group.ts:134]: i see, i don't get the dups (lets say `1463`) but I do get dups in group ids. Will fix that


https://github.com/hicommonwealth/commonwealth/assets/51641047/f988fb96-5157-4e7e-804f-c5f691dea270

Comment (rbennettcw) [reply packages/commonwealth/server/controllers/server_groups_methods/update_group.ts:134]: Right, that's what I meant 👍

File: packages/commonwealth/client/scripts/views/pages/Community/common/constants.ts [hunk 3]
```diff
@@ -35,23 +33,18 @@ export const requirementTypes = [
...
   .filter((x) => chainIdsToFind.includes(x.id.toLowerCase()));
-export const chainTypes = chainIdsToFind.map((x, index) => ({
-  value:
-    foundChains.find((y) => y.id.toLowerCase() === x)?.ChainNode?.ethChainId ||
-    index - 999, // TODO: some chains don't have an ethChainId
+export const chainTypes = chainIdsToFind.map((x) => ({
+  chainBase: cosmosBaseChainIds.includes(x) ? 'cosmos' : 'ethereum',
+  value: cosmosBaseChainIds.includes(x)
+    ? 'cosmos' // cosmos and injective have 'cosmos' id
+    : // TODO: polkadot doesn't have an ethChainId
```
Comment (lzach83) [inline packages/commonwealth/client/scripts/views/pages/Community/common/constants.ts:46]: Is this TODO complete? Makes this line look awkward if it is.

File: packages/commonwealth/server/controllers/server_groups_methods/update_group.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (CowMuon) [reply packages/commonwealth/server/controllers/server_groups_methods/update_group.ts]: mark as resolved pls

**Review by lzach83 — CHANGES_REQUESTED** [review]
Resolve my comment and what @rbennettcw referred to and everything else looks good! Great job

File: packages/commonwealth/client/scripts/views/pages/Community/common/constants.ts (hunk 3 above)
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/Community/common/constants.ts]: oh yes its complete. Removed the comment

File: packages/commonwealth/server/controllers/server_groups_methods/update_group.ts (hunk 1 above)
Comment (mzparacha) [reply packages/commonwealth/server/controllers/server_groups_methods/update_group.ts]: Fixed [here](https://github.com/hicommonwealth/commonwealth/pull/5499/commits/ba0b93468e64b57b3c883137dc769f25d2eaede1)

**Comment by mzparacha** [comment]
Addressed the comments above, thanks. And also ready for another review.

**Review by rbennettcw — APPROVED** [review]

**Review by lzach83 — APPROVED** [review]
lgtm

**Pull request merged**